packages/gestalt-codemods/x.x.x/flex-update-from-row.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Converts
+ *  <Row gap={x} />
+ * to
+ *  <Flex alignItems="center" justifyContent="start" gap={2x} />
+
+ - Double "gap" value, if present
+ - Add previous default value for alignItems, if not present
+ - Add previous default value for justifyContent, if not present
+ */
+export default function transformer(file, api) {
+  const j = api.jscodeshift;
+  const src = j(file.source);
+  let touchableLocalIdentifierName;
+
+  src.find(j.ImportDeclaration).forEach((path) => {
+    const decl = path.node;
+    if (decl.source.value !== 'gestalt') {
+      return;
+    }
+    const touchableSpecifier = decl.specifiers.find(
+      (node) => node.imported.name === 'Row'
+    );
+
+    if (!touchableSpecifier) {
+      return;
+    }
+
+    touchableLocalIdentifierName =
+      touchableSpecifier && touchableSpecifier.local.name;
+
+    const newSpecifiers = [
+      // Strip out Touchable import
+      ...decl.specifiers.filter((node) => node.imported.name !== 'Row'),
+      // Only add the new TapArea import if it is not already imported
+      decl.specifiers.every((node) => node.imported.name !== 'Flex') &&
+        j.importSpecifier(j.identifier('Flex')),
+    ].filter(Boolean);
+
+    // Sort all the imports alphabetically
+    newSpecifiers.sort((a, b) =>
+      a.imported.name.localeCompare(b.imported.name)
+    );
+
+    const newNode = j.importDeclaration(newSpecifiers, j.literal('gestalt'));
+
+    j(path).replaceWith(newNode);
+  });
+
+  if (!touchableLocalIdentifierName) {
+    // not imported
+    return null;
+  }
+
+  let hasModifications = false;
+
+  const transform = src
+    .find(j.JSXElement)
+    .forEach((path) => {
+      const { node } = path;
+
+      if (node.openingElement.name.name !== touchableLocalIdentifierName) {
+        return;
+      }
+
+      // Double 'gap'
+      node.openingElement.attributes = node.openingElement.attributes.map(
+        (attr) => {
+          const attribute = attr;
+          if (attribute.name && attribute.name.name === 'gap') {
+            const doubledVal = attribute.value.expression.value * 2;
```

Comment:
@dangerismycat What if the value of `gap` is not a number?

```jsx
const gap = 2;
//
<Row gap={gap} />
```

In those cases, we usually throw an exception to modify it manually: https://git.io/Jklij